### DISCUSSION TITLE
Docker does not listen on IPv6 connections

### DISCUSSION THREAD

**Discussion by MaZe3D** [dani-garcia/vaultwarden#3306] [Q&A] [answered]
### Subject of the issue

I am migrating my network to IPv6 only operation. Target of this is that my reverse proxy only accesses my services with a IPv6 only connection. While having great success migrating most of my containers (like mariadb, postgresql, nextcloud and some others), vaultwarden is the only one that won't listen on IPv6-addresses. I tested the connection to the container, both IPv4 and Ipv6 are pingable, but the web-interface is only reachable via IPv4. 

* Install method: Docker Image

* Other relevant details:
Container running with a IPv6 Address directly assigned from a ipv6-capable network.

### Steps to reproduce

1. Run the container with a IPv6 address assigned.
2. Try to connect to the web-interface with this address
3. You cannot access the web-interface
### Expected behaviour

The Web-Interface should be accessible from a docker with IPv6 address.
### Actual behaviour

The Web-Interface is not accessible via IPv6 address.

### Troubleshooting data

I can ping the container with an IPv6-Address assigned. My Docker command to run it:
```
docker run
  -d
  --name='vaultwarden'
  --net='br0.69'
  --ip='10.69.69.69'
  --ip6='fd00:59fb:c27b:3e0e:d6c6:ad5d:2b33:b44b'
  -e TZ="OBFUSCATED"
  -e HOST_OS="OBFUSCATED"
  -e HOST_HOSTNAME="OBFUSCATED"
  -e HOST_CONTAINERNAME="vaultwarden"
  -e 'TCP_PORT_80'='80'
  -e 'SERVER_ADMIN_EMAIL'='OBFUSCATED'
  -e 'SIGNUPS_ALLOWED'='true'
  -e 'INVITATIONS_ALLOWED'='true'
  -e 'WEBSOCKET_ENABLED'='true'
  -e 'ADMIN_TOKEN'='OBFUSCATED'
  -v 'OBFUSCATED':'/data':'rw' 'vaultwarden/server:latest'
```
(I replaced both IPv4 and IPv6 with dummy addresses and obfuscated everything that is unnecessary with `OBFUSCATED`).

Thanks for reading and have a lovely day!

**Comment by BlackDex** [comment, marked as answer]
You need to set te following variables to get it to work.
```bash
ROCKET_ADDRESS=::0
WEBSOCKET_ADDRESS=::0
```

**Reply by MaZe3D** [reply]
Thank you very much. Great to see that this is possible. And so easy.

**Reply by talha5389** [reply]
Thanks. After spending 5 hours debugging why my kubernetes pod wasn't working, I finally found that vaultwarden was only listening on ipv4

**Reply by talha5389** [reply]
@BlackDex how to listen on both v4 and v6?

**Reply by BlackDex** [reply]
If I'm correct from the top of my head you need to set it to `::` no numbers or dots, just two colons.
